keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/events/artifacts.kt

```diff
@@ -0,0 +1,27 @@
+package com.netflix.spinnaker.keel.api.events
+
+import com.netflix.spinnaker.keel.api.artifacts.DeliveryArtifact
+import com.netflix.spinnaker.keel.api.artifacts.SpinnakerArtifact
+
+/**
+ * An event that conveys information about one or more software artifacts that are
+ * potentially relevant to keel.
+ */
+data class ArtifactEvent(
+  val artifacts: List<SpinnakerArtifact>,
+  val details: Map<String, Any>?
+)
+
+/**
+ * Event emitted with a new [DeliveryArtifact] is registered.
+ */
+data class ArtifactRegisteredEvent(
+  val artifact: DeliveryArtifact
```

Comment:
What is the difference here between `SpinnakerArtifact` and `DeliveryArtifact`?

Reply:
`SpinnakerArtifact`, as the code docs in that class explain, is just a copycat of `com.netflix.spinnaker.core.models.Artifact`, i.e. the object that other Spinnaker services use to refer to a (versioned) artifact. `DeliveryArtifact` is a keel-specific model which describes high-level details about an artifact (as specified in the delivery config), so we can go find the actual artifacts in other Spinnaker services or other systems.